emitter.go

```diff
@@ -71,110 +71,123 @@ func CopyMulty(src io.Reader, writers ...io.Writer) (err error) {
-				}
-				continue
+			if er == io.EOF {
+				break Loop
 			}
-			requestID := string(meta[1])
-
-			if nr >= 5*1024*1024 {
-				log.Println("INFO: Large packet... We received ", len(payload), " bytes from ", src)
+			if er != nil {
+				break Loop
 			}
 
-			if Settings.debug {
-				Debug("[EMITTER] input:", string(payload[0:_maxN]), nr, "from:", src)
+			_maxN := nr
+			if nr > 500 {
+				_maxN = 500
 			}
+			if nr > 0 && len(buf) > nr {
+				payload := buf[:nr]
+				meta := payloadMeta(payload)
+				if len(meta) < 3 {
+					if Settings.debug {
+						Debug("[EMITTER] Found malformed record", string(payload[0:_maxN]), nr, "from:", src)
+					}
+					continue
+				}
+				requestID := string(meta[1])
 
-			if modifier != nil {
-				if isRequestPayload(payload) {
-					headSize := bytes.IndexByte(payload, '\n') + 1
-					body := payload[headSize:]
-					originalBodyLen := len(body)
-					body = modifier.Rewrite(body)
+				if nr >= 5*1024*1024 {
+					log.Println("INFO: Large packet... We received ", len(payload), " bytes from ", src)
+				}
 
-					// If modifier tells to skip request
-					if len(body) == 0 {
-						filteredRequests[requestID] = time.Now()
-						continue
-					}
+				if Settings.debug {
+					Debug("[EMITTER] input:", string(payload[0:_maxN]), nr, "from:", src)
+				}
 
-					if originalBodyLen != len(body) {
-						payload = append(payload[:headSize], body...)
+				if modifier != nil {
+					if isRequestPayload(payload) {
+						headSize := bytes.IndexByte(payload, '\n') + 1
+						body := payload[headSize:]
+						originalBodyLen := len(body)
+						body = modifier.Rewrite(body)
+
+						// If modifier tells to skip request
+						if len(body) == 0 {
+							filteredRequests[requestID] = time.Now()
+							continue
+						}
+
+						if originalBodyLen != len(body) {
+							payload = append(payload[:headSize], body...)
+						}
+
+						if Settings.debug {
+							Debug("[EMITTER] Rewritten input:", len(payload), "First 500 bytes:", string(payload[0:_maxN]))
+						}
+					} else {
+						if _, ok := filteredRequests[requestID]; ok {
+							delete(filteredRequests, requestID)
+							continue
+						}
 					}
+				}
 
-					if Settings.debug {
-						Debug("[EMITTER] Rewritten input:", len(payload), "First 500 bytes:", string(payload[0:_maxN]))
-					}
-				} else {
-					if _, ok := filteredRequests[requestID]; ok {
-						delete(filteredRequests, requestID)
+				if Settings.prettifyHTTP {
+					payload = prettifyHTTP(payload)
+					if len(payload) == 0 {
 						continue
 					}
 				}
-			}
-
-			if Settings.prettifyHTTP {
-				payload = prettifyHTTP(payload)
-				if len(payload) == 0 {
-					continue
-				}
-			}
 
-			if Settings.splitOutput {
-				// Simple round robin
-				if _, err := writers[wIndex].Write(payload); err != nil {
-					return err
-				}
+				if Settings.splitOutput {
+					// Simple round robin
+					if _, err := writers[wIndex].Write(payload); err != nil {
+						break Loop
+					}
```

Comment:
err is not handled

Reply:
Done

Reply:
It was handled outside. But I reverted break with a label.